app/src/main/java/com/infomaniak/drive/data/cache/FileMigration.kt

```diff
@@ -127,6 +131,154 @@ class FileMigration : RealmMigration {
+
+            // ShareLink migration
+            val shareLinkSchema = schema.create(ShareLink::class.java.simpleName).apply {
+                addField(ShareLink::url.name, String::class.java, FieldAttribute.REQUIRED)
+                addField(ShareLink::_right.name, String::class.java, FieldAttribute.REQUIRED)
+                addField(ShareLink::validUntil.name, Date::class.java)
+                addRealmObjectField(ShareLink::capabilities.name, shareLinkCapabilities)
+            }
+
+            // FileCategory migration
+            schema.get(FileCategory::class.java.simpleName)?.apply {
+                renameField("id", FileCategory::categoryId.name)
+                renameField("iaCategoryUserValidation", FileCategory::userValidation.name)
+                renameField("isGeneratedByIa", FileCategory::isGeneratedByAI.name)
+                renameField("addedToFileAt", FileCategory::addedAt.name)
+            }
+
+            // File migration
+            schema.get(File::class.java.simpleName)?.apply {
+                removeField("collaborativeFolder")
+                removeField("onlyofficeConvertExtension")
+                removeField("hasVersion")
+                removeField("lastAccessedAt")
+                removeField("nbVersion")
+                removeField("sizeWithVersions")
+                removeField("shareLink")
+
+                renameField("convertedType", File::extensionType.name)
+                renameField("createdAt", File::addedAt.name)
+                renameField("fileCreatedAt", File::createdAt.name)
+                renameField("nameNaturalSorting", File::sortedName.name)
+                renameField("onlyoffice", File::hasOnlyoffice.name)
+
+                addField(File::parentId.name, Int::class.java, FieldAttribute.REQUIRED)
+                addRealmObjectField(File::conversion.name, fileConversionSchema)
+                addRealmObjectField(File::dropbox.name, dropboxSchema)
+                addRealmObjectField(File::version.name, fileVersionSchema)
+                addRealmObjectField(File::sharelink.name, shareLinkSchema)
+            }
+
+            // FileActivity migration
+            schema.get(FileActivity::class.java.simpleName)?.apply {
+                removeField("path")
+            }
+
+            // Set embedded objects
+            schema.get(ShareLinkCapabilities::class.java.simpleName)?.isEmbedded = true
+            schema.get(ShareLink::class.java.simpleName)?.isEmbedded = true
+            schema.get(DropBoxValidity::class.java.simpleName)?.isEmbedded = true
+            schema.get(DropBoxSize::class.java.simpleName)?.isEmbedded = true
+            schema.get(DropBoxCapabilities::class.java.simpleName)?.isEmbedded = true
+            schema.get(DropBox::class.java.simpleName)?.isEmbedded = true
+            schema.get(FileVersion::class.java.simpleName)?.isEmbedded = true
+            schema.get(FileConversion::class.java.simpleName)?.isEmbedded = true
+        }
```

Comment:
```suggestion
            oldVersionTemp++
        }
```

Reply:
I think it's better not to put it, so it's at the next migration that you have to add it, so you get rid of the warnings